# **Tourism in Ireland** <br>Analysis of Pedestrian Footfall in Dublin City Center
### *Prepared by Ben Wilson, sbs24004*

In [1]:
import pandas as pd
import seaborn as sns

**1. Data Cleaning and Preparation**

In [2]:
footfall23_df = pd.read_csv('DCCpfc2023.csv')

In [3]:
footfall23_df.head(5)

Time  Aston Quay/Fitzgeralds  Aston Quay/Fitzgeralds IN  \
0  01/01/2023 00:00                  3700.0                     1455.0   
1  01/01/2023 01:00                  4369.0                     2142.0   
2  01/01/2023 02:00                  3807.0                     2031.0   
3  01/01/2023 03:00                  2477.0                     1352.0   
4  01/01/2023 04:00                  1534.0                      836.0   

   Aston Quay/Fitzgeralds OUT  Bachelors walk/Bachelors way  \
0                      2245.0                           0.0   
1                      2227.0                           0.0   
2                      1776.0                           0.0   
3                      1125.0                           0.0   
4                       698.0                           0.0   

   Bachelors walk/Bachelors way IN  Bachelors walk/Bachelors way OUT  \
0                              0.0                               0.0   
1                              0.0                               0.0   
2                              0.0                               0.0   
3                              0.0                               0.0   
4                              0.0                               0.0   

   Baggot st lower/Wilton tce inbound  \
0                                74.0   
1                               100.0   
2                                52.0   
3                                42.0   
4                                11.0   

   Baggot st lower/Wilton tce inbound Pedestrians IN  \
0                                               40.0   
1                                               46.0   
2                                               15.0   
3                                               18.0   
4                                                1.0   

   Baggot st lower/Wilton tce inbound Pedestrians OUT  ...  \
0                                               34.0   ...   
1                                               54.0   ...   
2                                               37.0   ...   
3                                               24.0   ...   
4                                               10.0   ...   

   Talbot st/Guineys OUT  Talbot st/Murrays Pharmacy  \
0                 1089.0                      2891.0   
1                  730.0                      2562.0   
2                  379.0                      2207.0   
3                  350.0                      1834.0   
4                  345.0                      1307.0   

   Talbot st/Murrays Pharmacy IN  Talbot st/Murrays Pharmacy OUT  \
0                         2489.0                           402.0   
1                         2263.0                           299.0   
2                         2116.0                            91.0   
3                         1747.0                            87.0   
4                         1236.0                            71.0   

   Westmoreland Street East/Fleet street  \
0                                 1011.0   
1                                  634.0   
2                                  501.0   
3                                  445.0   
4                                  127.0   

   Westmoreland Street East/Fleet street IN  \
0                                     397.0   
1                                     381.0   
2                                     340.0   
3                                     306.0   
4                                      94.0   

   Westmoreland Street East/Fleet street OUT  \
0                                      614.0   
1                                      253.0   
2                                      161.0   
3                                      139.0   
4                                       33.0   

   Westmoreland Street West/Carrolls  Westmoreland Street West/Carrolls IN  \
0                             1835.0                                1256.0   
1                             1426.0                          

In [4]:
footfall23_df.shape

(8760, 115)

8760 hours in a Year, 38 counters x 3 = 114 + Time?

In [5]:
footfall23_df.describe(include='all')

Time  Aston Quay/Fitzgeralds  Aston Quay/Fitzgeralds IN  \
count               8760             8759.000000                8759.000000   
unique              8759                     NaN                        NaN   
top     26/03/2023 02:00                     NaN                        NaN   
freq                   2                     NaN                        NaN   
mean                 NaN             3371.344446                1633.542528   
std                  NaN             1808.971110                 853.872352   
min                  NaN               27.000000                   6.000000   
25%                  NaN             1900.500000                1010.500000   
50%                  NaN             3685.000000                1712.000000   
75%                  NaN             4583.000000                2190.500000   
max                  NaN            10658.000000                5663.000000   

        Aston Quay/Fitzgeralds OUT  Bachelors walk/Bachelors way  \
count                  8759.000000                   7396.000000   
unique                         NaN                           NaN   
top                            NaN                           NaN   
freq                           NaN                           NaN   
mean                   1737.801918                   1408.451190   
std                     978.887637                   3165.336185   
min                      21.000000                      0.000000   
25%                     876.500000                      0.000000   
50%                    1947.000000                     62.500000   
75%                    2423.000000                   1605.000000   
max                    4995.000000                  16703.000000   

        Bachelors walk/Bachelors way IN  Bachelors walk/Bachelors way OUT  \
count                       7396.000000                       7396.000000   
unique                              NaN                               NaN   
top                                 NaN                               NaN   
freq                                NaN                               NaN   
mean                         973.738778                        434.712412   
std                         2253.567648                       1053.472648   
min                            0.000000                          0.000000   
25%                            0.000000                          0.000000   
50%                           50.000000                         11.000000   
75%                         1071.250000                        499.000000   
max                        15410.000000                       7400.000000   

        Baggot st lower/Wilton tce inbound  \
count                          8759.000000   
unique                                 NaN   
top                                    NaN   
freq                                   NaN   
mean                            124.538760   
std                             135.577782   
min                               0.000000   
25%                              16.000000   
50%                              87.000000   
75%                             187.000000   
max                            1755.000000   

        Baggot st lower/Wilton tce inbound Pedestrians IN  \
count                                         8759.000000   
unique                                                NaN   
top                                                   NaN   
freq                                                  NaN   
mean                                            75.382235   
std                                             90.627446   
min                                              0.000000   
25%                                              6.000000   
50%                                             50.000000   
75%                                            113.000000   
max                                           1732.000000   

        Baggot st lower/Wilton tce inbound 

In [6]:
footfall23_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8760 entries, 0 to 8759
Columns: 115 entries, Time to Westmoreland Street West/Carrolls OUT
dtypes: float64(114), object(1)
memory usage: 7.7+ MB


In [7]:
footfall23_df.columns

Index(['Time', 'Aston Quay/Fitzgeralds', 'Aston Quay/Fitzgeralds IN',
       'Aston Quay/Fitzgeralds OUT', 'Bachelors walk/Bachelors way',
       'Bachelors walk/Bachelors way IN', 'Bachelors walk/Bachelors way OUT',
       'Baggot st lower/Wilton tce inbound',
       'Baggot st lower/Wilton tce inbound Pedestrians IN',
       'Baggot st lower/Wilton tce inbound Pedestrians OUT',
       ...
       'Talbot st/Guineys OUT', 'Talbot st/Murrays Pharmacy',
       'Talbot st/Murrays Pharmacy IN', 'Talbot st/Murrays Pharmacy OUT',
       'Westmoreland Street East/Fleet street',
       'Westmoreland Street East/Fleet street IN',
       'Westmoreland Street East/Fleet street OUT',
       'Westmoreland Street West/Carrolls',
       'Westmoreland Street West/Carrolls IN',
       'Westmoreland Street West/Carrolls OUT'],
      dtype='object', length=115)

**1.1** <br>I want to remove columns with IN or OUT, and leave only totals. Create user defined function, remove_IN_OUT()

In [8]:
from UserDefinedFunctions import remove_IN_OUT

In [9]:
footfall23_df = remove_IN_OUT(footfall23_df)

In [10]:
footfall23_df.shape

(8760, 39)

Colums removed, 38 locations and time

In [11]:
list(footfall23_df)

['Time',
 'Aston Quay/Fitzgeralds',
 'Bachelors walk/Bachelors way',
 'Baggot st lower/Wilton tce inbound',
 'Baggot st upper/Mespil rd/Bank',
 'Capel st/Mary street',
 'College Green/Bank Of Ireland',
 'College Green/Church Lane',
 'College st/Westmoreland st',
 "D'olier st/Burgh Quay",
 'Dame Street/Londis',
 'Dawson Street',
 'Dawson Street old',
 'Dawson Street/Molesworth Pedestrian',
 'Grafton st/Monsoon',
 'Grafton Street / Nassau Street / Suffolk Street',
 'Grafton Street/CompuB',
 'Grand Canal st upp/Clanwilliam place',
 'Grand Canal st upp/Clanwilliam place/Google',
 'Henry Street/Coles Lane/Dunnes',
 'Liffey st/Halfpenny Bridge ',
 'Liffey Street old',
 'Mary st/Jervis st',
 'Newcomen Bridge/Charleville mall inbound',
 'Newcomen Bridge/Charleville mall outbound',
 'North Wall Quay/Samuel Beckett bridge East',
 'North Wall Quay/Samuel Beckett bridge West',
 "O'Connell St/Parnell St/AIB",
 "O'Connell St/Pennys Pedestrian",
 "O'Connell st/Princes st North",
 "O'Connell Street Pe

With IN and OUT columns removed, reexplore Data Set

In [12]:
footfall23_df.describe(include='all')

Time  Aston Quay/Fitzgeralds  \
count               8760             8759.000000   
unique              8759                     NaN   
top     26/03/2023 02:00                     NaN   
freq                   2                     NaN   
mean                 NaN             3371.344446   
std                  NaN             1808.971110   
min                  NaN               27.000000   
25%                  NaN             1900.500000   
50%                  NaN             3685.000000   
75%                  NaN             4583.000000   
max                  NaN            10658.000000   

        Bachelors walk/Bachelors way  Baggot st lower/Wilton tce inbound  \
count                    7396.000000                         8759.000000   
unique                           NaN                                 NaN   
top                              NaN                                 NaN   
freq                             NaN                                 NaN   
mean                     1408.451190                          124.538760   
std                      3165.336185                          135.577782   
min                         0.000000                            0.000000   
25%                         0.000000                           16.000000   
50%                        62.500000                           87.000000   
75%                      1605.000000                          187.000000   
max                     16703.000000                         1755.000000   

        Baggot st upper/Mespil rd/Bank  Capel st/Mary street  \
count                      8759.000000           8759.000000   
unique                             NaN                   NaN   
top                                NaN                   NaN   
freq                               NaN                   NaN   
mean                       4159.843019           1235.841877   
std                        3044.631578           1037.122022   
min                           4.000000              4.000000   
25%                        1310.000000            387.000000   
50%                        4039.000000           1018.000000   
75%                        6225.000000           1834.000000   
max                       15100.000000           9217.000000   

        College Green/Bank Of Ireland  College Green/Church Lane  \
count                     8705.000000                7036.000000   
unique                            NaN                        NaN   
top                               NaN                        NaN   
freq                              NaN                        NaN   
mean                       545.744515                 590.919983   
std                        631.544924                 589.564217   
min                          0.000000                   0.000000   
25%                        116.000000                  22.000000   
50%                        375.000000                 408.000000   
75%                        687.000000                1086.250000   
max                       3919.000000                2316.000000   

        College st/Westmoreland st  D'olier st/Burgh Quay  ...  \
count                  8759.000000            8759.000000  ...   
unique                         NaN                    NaN  ...   
top                            NaN                    NaN  ...   
freq                           NaN                    NaN  ...   
mean                    602.385775             941.580203  ...   
std                     462.085500             545.537413  ...   
min                       1.000000               1.000000  ...   
25%                     168.000000             514.000000  ...   
50%                     537.000000            1025.000000  ...   
75%                     995.000000            1304.000000  ...   
max                    3372.000000            3632.000000  ...   

        O'Connell st/Princes st North  \
count                     8759.000000   
unique               

A lot of missing values

In [13]:
footfall23_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8760 entries, 0 to 8759
Data columns (total 39 columns):
 #   Column                                                    Non-Null Count  Dtype  
---  ------                                                    --------------  -----  
 0   Time                                                      8760 non-null   object 
 1   Aston Quay/Fitzgeralds                                    8759 non-null   float64
 2   Bachelors walk/Bachelors way                              7396 non-null   float64
 3   Baggot st lower/Wilton tce inbound                        8759 non-null   float64
 4   Baggot st upper/Mespil rd/Bank                            8759 non-null   float64
 5   Capel st/Mary street                                      8759 non-null   float64
 6   College Green/Bank Of Ireland                             8705 non-null   float64
 7   College Green/Church Lane                                 7036 non-null   float64
 8   College st/Westmor

**1.2** <br> I want to change the index of the data frame to time, the first column

In [14]:
footfall23_df['Time'].head(5)

0    01/01/2023 00:00
1    01/01/2023 01:00
2    01/01/2023 02:00
3    01/01/2023 03:00
4    01/01/2023 04:00
Name: Time, dtype: object

First I need to convert to a day time object

In [15]:
footfall23_df['Time'] = pd.to_datetime(footfall23_df['Time'], format='%d/%m/%Y %H:%M') # Convert to dateTime format

In [16]:
footfall23_df['Time'].head(5)

0   2023-01-01 00:00:00
1   2023-01-01 01:00:00
2   2023-01-01 02:00:00
3   2023-01-01 03:00:00
4   2023-01-01 04:00:00
Name: Time, dtype: datetime64[ns]

In [17]:
footfall23_df.set_index('Time', inplace=True) # Set index to Time

In [18]:
footfall23_df.head(5)

Aston Quay/Fitzgeralds  Bachelors walk/Bachelors way  \
Time                                                                        
2023-01-01 00:00:00                  3700.0                           0.0   
2023-01-01 01:00:00                  4369.0                           0.0   
2023-01-01 02:00:00                  3807.0                           0.0   
2023-01-01 03:00:00                  2477.0                           0.0   
2023-01-01 04:00:00                  1534.0                           0.0   

                     Baggot st lower/Wilton tce inbound  \
Time                                                      
2023-01-01 00:00:00                                74.0   
2023-01-01 01:00:00                               100.0   
2023-01-01 02:00:00                                52.0   
2023-01-01 03:00:00                                42.0   
2023-01-01 04:00:00                                11.0   

                     Baggot st upper/Mespil rd/Bank  Capel st/Mary street  \
Time                                                                        
2023-01-01 00:00:00                          4296.0                 808.0   
2023-01-01 01:00:00                          1466.0                 639.0   
2023-01-01 02:00:00                          1642.0                 546.0   
2023-01-01 03:00:00                           967.0                 349.0   
2023-01-01 04:00:00                           328.0                 309.0   

                     College Green/Bank Of Ireland  College Green/Church Lane  \
Time                                                                            
2023-01-01 00:00:00                         1325.0                      829.0   
2023-01-01 01:00:00                         1209.0                      685.0   
2023-01-01 02:00:00                         1151.0                      589.0   
2023-01-01 03:00:00                         1368.0                      556.0   
2023-01-01 04:00:00                          710.0                      207.0   

                     College st/Westmoreland st  D'olier st/Burgh Quay  \
Time                                                                     
2023-01-01 00:00:00                       801.0                 1714.0   
2023-01-01 01:00:00                       641.0                 1411.0   
2023-01-01 02:00:00                       481.0                 1219.0   
2023-01-01 03:00:00                       395.0                 1225.0   
2023-01-01 04:00:00                       110.0                  559.0   

                     Dame Street/Londis  ...  O'Connell st/Princes st North  \
Time                                     ...                                  
2023-01-01 00:00:00               858.0  ...                         1408.0   
2023-01-01 01:00:00               574.0  ...                          891.0   
2023-01-01 02:00:00               664.0  ...                          870.0   
2023-01-01 03:00:00               771.0  ...                          722.0   
2023-01-01 04:00:00               337.0  ...                          283.0   

                     O'Connell Street Pennys - PYRO EVO Temporary Counter  \
Time                                                                        
2023-01-01 00:00:00                                                NaN      
2023-01-01 01:00:00                                                NaN      
2023-01-01 02:00:00                                                NaN      
2023-01-01 03:00:00                                                NaN      
2023-01-01 04:00:00                                                NaN      

                     Phibsborough Rd/Enniskerry Road  \
Time                                                   
2023-01-01 00:00:00                             56.0   
2023-01-01 01:00:00                             86.0   
2023-01-01 02:00:00                             63.0   
2023-01-01 03:00:00                             90.0   
2023-01-01

In [19]:
footfall23_df.describe(include='all')

Aston Quay/Fitzgeralds  Bachelors walk/Bachelors way  \
count             8759.000000                   7396.000000   
mean              3371.344446                   1408.451190   
std               1808.971110                   3165.336185   
min                 27.000000                      0.000000   
25%               1900.500000                      0.000000   
50%               3685.000000                     62.500000   
75%               4583.000000                   1605.000000   
max              10658.000000                  16703.000000   

       Baggot st lower/Wilton tce inbound  Baggot st upper/Mespil rd/Bank  \
count                         8759.000000                     8759.000000   
mean                           124.538760                     4159.843019   
std                            135.577782                     3044.631578   
min                              0.000000                        4.000000   
25%                             16.000000                     1310.000000   
50%                             87.000000                     4039.000000   
75%                            187.000000                     6225.000000   
max                           1755.000000                    15100.000000   

       Capel st/Mary street  College Green/Bank Of Ireland  \
count           8759.000000                    8705.000000   
mean            1235.841877                     545.744515   
std             1037.122022                     631.544924   
min                4.000000                       0.000000   
25%              387.000000                     116.000000   
50%             1018.000000                     375.000000   
75%             1834.000000                     687.000000   
max             9217.000000                    3919.000000   

       College Green/Church Lane  College st/Westmoreland st  \
count                7036.000000                 8759.000000   
mean                  590.919983                  602.385775   
std                   589.564217                  462.085500   
min                     0.000000                    1.000000   
25%                    22.000000                  168.000000   
50%                   408.000000                  537.000000   
75%                  1086.250000                  995.000000   
max                  2316.000000                 3372.000000   

       D'olier st/Burgh Quay  Dame Street/Londis  ...  \
count            8759.000000         8759.000000  ...   
mean              941.580203          449.773718  ...   
std               545.537413          345.492423  ...   
min                 1.000000            0.000000  ...   
25%               514.000000          145.000000  ...   
50%              1025.000000          407.000000  ...   
75%              1304.000000          678.000000  ...   
max              3632.000000         1792.000000  ...   

       O'Connell st/Princes st North  \
count                    8759.000000   
mean                      710.051262   
std                       499.244161   
min                         3.000000   
25%                       244.000000   
50%                       703.000000   
75%                      1072.000000   
max                      2625.000000   

       O'Connell Street Pennys - PYRO EVO Temporary Counter  \
count                                                0.0      
mean                                                 NaN      
std                                                  NaN      
min                                                  NaN      
25%                                                  NaN      
50%                                                  NaN      
75%                                                  NaN      
max                                                  NaN      

       Phibsborough Rd/Enniskerry Road  \
count                      8759.000000   
mean                         95.790501   
std                          70.980989  

In [20]:
footfall23_df.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 8760 entries, 2023-01-01 00:00:00 to 2023-12-31 23:00:00
Data columns (total 38 columns):
 #   Column                                                    Non-Null Count  Dtype  
---  ------                                                    --------------  -----  
 0   Aston Quay/Fitzgeralds                                    8759 non-null   float64
 1   Bachelors walk/Bachelors way                              7396 non-null   float64
 2   Baggot st lower/Wilton tce inbound                        8759 non-null   float64
 3   Baggot st upper/Mespil rd/Bank                            8759 non-null   float64
 4   Capel st/Mary street                                      8759 non-null   float64
 5   College Green/Bank Of Ireland                             8705 non-null   float64
 6   College Green/Church Lane                                 7036 non-null   float64
 7   College st/Westmoreland st                                8759 non-nu

**1.3** <br> Check for missing values

In [21]:
# Check for missing values
print("Are there missing values?", footfall23_df.isnull().values.any())

Are there missing values? True


In [22]:
# How many missing observations per column/street sensor?
print(footfall23_df.isnull().sum())

Aston Quay/Fitzgeralds                                         1
Bachelors walk/Bachelors way                                1364
Baggot st lower/Wilton tce inbound                             1
Baggot st upper/Mespil rd/Bank                                 1
Capel st/Mary street                                           1
College Green/Bank Of Ireland                                 55
College Green/Church Lane                                   1724
College st/Westmoreland st                                     1
D'olier st/Burgh Quay                                          1
Dame Street/Londis                                             1
Dawson Street                                               8760
Dawson Street old                                           8760
Dawson Street/Molesworth Pedestrian                         8472
Grafton st/Monsoon                                             1
Grafton Street / Nassau Street / Suffolk Street              136
Grafton Street/CompuB    

There are some streets with almost or completely missing values. Before I remove them I want to check of theyre of the time period I am interested in, which I will be focussing on St.Patrick day holiday, March 17th

1.3.1
patricks day close up

In [23]:
# Create a new data frame on St.Pats day
patrick23_df = footfall23_df[(footfall23_df.index.month == 3) & (footfall23_df.index.day == 17)]

In [24]:
patrick23_df.head(5)

Aston Quay/Fitzgeralds  Bachelors walk/Bachelors way  \
Time                                                                        
2023-03-17 00:00:00                   837.0                           NaN   
2023-03-17 01:00:00                  2178.0                           NaN   
2023-03-17 02:00:00                  1719.0                           NaN   
2023-03-17 03:00:00                  1422.0                           NaN   
2023-03-17 04:00:00                  1373.0                           NaN   

                     Baggot st lower/Wilton tce inbound  \
Time                                                      
2023-03-17 00:00:00                                67.0   
2023-03-17 01:00:00                                50.0   
2023-03-17 02:00:00                                10.0   
2023-03-17 03:00:00                                 7.0   
2023-03-17 04:00:00                                 5.0   

                     Baggot st upper/Mespil rd/Bank  Capel st/Mary street  \
Time                                                                        
2023-03-17 00:00:00                          5469.0                1195.0   
2023-03-17 01:00:00                          2292.0                 949.0   
2023-03-17 02:00:00                          1835.0                 638.0   
2023-03-17 03:00:00                          1842.0                 100.0   
2023-03-17 04:00:00                           710.0                 195.0   

                     College Green/Bank Of Ireland  College Green/Church Lane  \
Time                                                                            
2023-03-17 00:00:00                          157.0                      513.0   
2023-03-17 01:00:00                           97.0                      308.0   
2023-03-17 02:00:00                          116.0                      280.0   
2023-03-17 03:00:00                           84.0                      244.0   
2023-03-17 04:00:00                           26.0                      111.0   

                     College st/Westmoreland st  D'olier st/Burgh Quay  \
Time                                                                     
2023-03-17 00:00:00                       414.0                 1150.0   
2023-03-17 01:00:00                       183.0                  890.0   
2023-03-17 02:00:00                       182.0                  943.0   
2023-03-17 03:00:00                       124.0                  664.0   
2023-03-17 04:00:00                        52.0                  321.0   

                     Dame Street/Londis  ...  O'Connell st/Princes st North  \
Time                                     ...                                  
2023-03-17 00:00:00               522.0  ...                          557.0   
2023-03-17 01:00:00               295.0  ...                          307.0   
2023-03-17 02:00:00               399.0  ...                          251.0   
2023-03-17 03:00:00               231.0  ...                          226.0   
2023-03-17 04:00:00                95.0  ...                          111.0   

                     O'Connell Street Pennys - PYRO EVO Temporary Counter  \
Time                                                                        
2023-03-17 00:00:00                                                NaN      
2023-03-17 01:00:00                                                NaN      
2023-03-17 02:00:00                                                NaN      
2023-03-17 03:00:00                                                NaN      
2023-03-17 04:00:00                                                NaN      

                     Phibsborough Rd/Enniskerry Road  \
Time                                                   
2023-03-17 00:00:00                             56.0   
2023-03-17 01:00:00                             31.0   
2023-03-17 02:00:00                             12.0   
2023-03-17 03:00:00                             20.0   
2023-03-17

In [25]:
# How many missing observations per column/street sensor on St.Patricks day?
print(patrick23_df.isnull().sum())

Aston Quay/Fitzgeralds                                       0
Bachelors walk/Bachelors way                                24
Baggot st lower/Wilton tce inbound                           0
Baggot st upper/Mespil rd/Bank                               0
Capel st/Mary street                                         0
College Green/Bank Of Ireland                                0
College Green/Church Lane                                    0
College st/Westmoreland st                                   0
D'olier st/Burgh Quay                                        0
Dame Street/Londis                                           0
Dawson Street                                               24
Dawson Street old                                           24
Dawson Street/Molesworth Pedestrian                         24
Grafton st/Monsoon                                           0
Grafton Street / Nassau Street / Suffolk Street              0
Grafton Street/CompuB                                  

Compare St.Pats missing data to full year missing data

In [26]:
# create data frames to store missing info
missing_patrick23 = patrick23_df.isnull().sum()
missing_footfall23 = footfall23_df.isnull().sum()

In [27]:
# Concatenate them horizontally
missing_comparison23 = pd.concat([missing_patrick23, missing_footfall23], axis=1)
# Column Names
missing_comparison23.columns = ['Missing in Patrick23_df', 'Missing in footfall23_df']

In [28]:
missing_comparison23.head(40)

Missing in Patrick23_df  \
Aston Quay/Fitzgeralds                                                    0   
Bachelors walk/Bachelors way                                             24   
Baggot st lower/Wilton tce inbound                                        0   
Baggot st upper/Mespil rd/Bank                                            0   
Capel st/Mary street                                                      0   
College Green/Bank Of Ireland                                             0   
College Green/Church Lane                                                 0   
College st/Westmoreland st                                                0   
D'olier st/Burgh Quay                                                     0   
Dame Street/Londis                                                        0   
Dawson Street                                                            24   
Dawson Street old                                                        24   
Dawson Street/Molesworth Pedestrian                                      24   
Grafton st/Monsoon                                                        0   
Grafton Street / Nassau Street / Suffolk Street                           0   
Grafton Street/CompuB                                                     0   
Grand Canal st upp/Clanwilliam place                                      0   
Grand Canal st upp/Clanwilliam place/Google                               0   
Henry Street/Coles Lane/Dunnes                                            0   
Liffey st/Halfpenny Bridge                                                0   
Liffey Street old                                                        24   
Mary st/Jervis st                                                         0   
Newcomen Bridge/Charleville mall inbound                                 24   
Newcomen Bridge/Charleville mall outbound                                24   
North Wall Quay/Samuel Beckett bridge East                                0   
North Wall Quay/Samuel Beckett bridge West                                0   
O'Connell St/Parnell St/AIB                                               0   
O'Connell St/Pennys Pedestrian                                           24   
O'Connell st/Princes st North                                             0   
O'Connell Street Pennys - PYRO EVO Temporary Co...                       24   
Phibsborough Rd/Enniskerry Road                                           0   
Phibsborough Rd/Munster St (Removed due to Over...                       24   
Richmond st south/Portabello Harbour inbound                              0   
Richmond st south/Portabello Harbour outbound                             0   
Talbot st/Guineys                                                         0   
Talbot st/Murrays Pharmacy                                               24   
Westmoreland Street East/Fleet street                                     0   
Westmoreland Street West/Carrolls                                        24   

                                                    Missing in footfall23_df  
Aston Quay/Fitzgeralds                                                     1  
Bachelors walk/Bachelors way                                            1364  
Baggot st lower/Wilton tce inbound                                         1  
Baggot st upper/Mespil rd/Bank                                             1  
Capel st/Mary street                                                       1  
College Green/Bank Of Ireland                                             55  
College Green/Church Lane                                               1724  
College st/Westmoreland st                                                 1  
D'olier st/Burgh Quay                                                      1  
Dame Street/Londis                                                         1  
Dawson Street                                                           8760  
Dawson Street old     

I can see from this table that the following streets are missing Data for both St.Patricks day, and most of the year. 
- A
- Dawson Street
- Dawson Street old
- Dawson Street/Molesworth Pedestrian
- Liffey Street old
- Newcomen Bridge/Charleville mall inbound
- Newcomen Bridge/Charleville mall outbound
- O'Connell Street Pennys - PYRO EVO Temporary Counter
- Phibsborough Rd/Munster St (Removed due to Overcounting)
- Talbot st/Murrays Pharmacy
- Westmoreland Street West/Carrolls

The following has Data missing for Pats, and only some of the year:
- B
- Bachelors walk/Bachelors way, 1364
- O'Connell St/Pennys Pedestrian, 2626

My thinking now is to remove above columns A, and possibly interpolate columns B, but probably remove all. First, I would like to visualize where the sensors are in Dublin, and what reduction in data remova lwould look like. There is also data from previous yeaars which could be useful for this investigation, howver for now Im going to focus on 2023.

**1.3.2 Visualizing sensor locations**

Inspect location file and visulaize

In [29]:
counterLocations_df = pd.read_csv('DCCcounterlocations.csv')
counterLocations_df.head(40)

_id                                  Counter Locations  \
0     1                                       Henry Street   
1     2                        O'Connell St Outside Clerys   
2     3                                        Mary Street   
3     4                                       Capel Street   
4     5                                         Aston Quay   
5     6                            Grafton Street @ CompuB   
6     7                                Talbot Street North   
7     8                         Doilier Street, Burgh Quay   
8     9                                        Dame Street   
9    10                                Talbot Street South   
10   11                     O'Connell St, Parnell St @ AIB   
11   12    Grafton Street / Nassau Street / Suffolk Street   
12   13                     College Green, Bank Of Ireland   
13   14                        O'Connell St Outside Pennys   
14   15                                     Grafton Street   
15   16                           Westmoreland Street East   
16   17                                      Dawson Street   
17   18                                      Liffey Street   
18   19                           Westmoreland Street West   
19   20                        College Green @ Church Lane   
20   21                       College Green - Dame St Side   
21   22                                     Bachelors Walk   
22   23                 Baggot st lower/Wilton tce inbound   
23   24                     Baggot st upper/Mespil rd/Bank   
24   25               Grand Canal st upp/Clanwilliam place   
25   26        Grand Canal st upp/Clanwilliam place/Google   
26   27           Newcomen Bridge/Charleville mall inbound   
27   28          Newcomen Bridge/Charleville mall outbound   
28   29         North Wall Quay/Samuel Beckett bridge East   
29   30         North Wall Quay/Samuel Beckett bridge West   
30   31                    Phibsborough Rd/Enniskerry Road   
31   32  Phibsborough Rd/Munster St (Removed due to Ove...   
32   33       Richmond st south/Portabello Harbour inbound   
33   34      Richmond st south/Portabello Harbour outbound   

                                     Eco-Visio Oupput   Latitude  Longitude  \
0                      Henry Street/Coles Lane/Dunnes  53.349730  -6.260900   
1                       O'Connell st/Princes st North  53.349020  -6.260050   
2                                   Mary st/Jervis st  53.348770  -6.266740   
3                                Capel st/Mary street  53.348420  -6.269180   
4                              Aston Quay/Fitzgeralds  53.346620  -6.259820   
5                               Grafton Street/CompuB  53.343370  -6.258980   
6                                   Talbot st/Guineys  53.350540  -6.255280   
7                               D'olier st/Burgh Quay  53.346900  -6.258720   
8                                  Dame Street/Londis  53.344240  -6.261160   
9                          Talbot st/Murrays Pharmacy  53.350120  -6.257700   
10                        O'Connell St/Parnell St/AIB  53.352317  -6.261675   
11    Grafton Street / Nassau Street / Suffolk Street  53.343520  -6.259120   
12                      College Green/Bank Of Ireland  53.345050  -6.259390   
13                     O'Connell St/Pennys Pedestrian  53.348740  -6.260110   
14                                 Grafton st/Monsoon  53.340820  -6.260350   
15              Westmoreland Street East/Fleet street  53.345540  -6.259190   
16                Dawson Street/Molesworth Pedestrian  53.341194  -6.258337   
17                        Liffey st/Halfpenny Bridge   53.346870  -6.263350   
18                  Westmoreland Street West/Carrolls  53.346348  -6.259197   
19                          College Green/Church Lane  53.344263  -6.260774   
20                         College st/Westmoreland st  53.345100  -6.258778   
21                       Bachelors walk/Bachelors way  53.347200  -6.260863   
22              

**1.3.2.1**
There are 38 streets with counts, but only 34 counter locations. I want to compare counter locations with the the footfall counts, to see where these unlisted sensors are

In [30]:
# create series to store Counter Locations
series_loc_counterLocations = pd.Series(counterLocations_df['Eco-Visio Oupput'])
series_loc_footfall23 = pd.Series(footfall23_df.columns)

# Convert Series to DataFrames
location_counterLocations = pd.DataFrame({'Counter Locations': series_loc_counterLocations})
location_footfall23 = pd.DataFrame({'Footfall Locations': series_loc_footfall23})

# Merge the DataFrames
location_comparison = pd.merge(location_counterLocations, location_footfall23, how='outer', left_on='Counter Locations', right_on='Footfall Locations')

# Sort the merged DataFrame alphabetically
location_comparison.sort_values(by='Counter Locations', inplace=True)

# Print the merged DataFrame
location_comparison

Counter Locations  \
4                              Aston Quay/Fitzgeralds   
21                       Bachelors walk/Bachelors way   
22                 Baggot st lower/Wilton tce inbound   
23                     Baggot st upper/Mespil rd/Bank   
3                                Capel st/Mary street   
12                      College Green/Bank Of Ireland   
19                          College Green/Church Lane   
20                         College st/Westmoreland st   
7                               D'olier st/Burgh Quay   
8                                  Dame Street/Londis   
16                Dawson Street/Molesworth Pedestrian   
11    Grafton Street / Nassau Street / Suffolk Street   
5                               Grafton Street/CompuB   
14                                 Grafton st/Monsoon   
24               Grand Canal st upp/Clanwilliam place   
25        Grand Canal st upp/Clanwilliam place/Google   
0                      Henry Street/Coles Lane/Dunnes   
17                        Liffey st/Halfpenny Bridge    
2                                   Mary st/Jervis st   
26           Newcomen Bridge/Charleville mall inbound   
27          Newcomen Bridge/Charleville mall outbound   
28         North Wall Quay/Samuel Beckett bridge East   
29         North Wall Quay/Samuel Beckett bridge West   
10                        O'Connell St/Parnell St/AIB   
13                     O'Connell St/Pennys Pedestrian   
1                       O'Connell st/Princes st North   
30                    Phibsborough Rd/Enniskerry Road   
31  Phibsborough Rd/Munster St (Removed due to Ove...   
32       Richmond st south/Portabello Harbour inbound   
33      Richmond st south/Portabello Harbour outbound   
6                                   Talbot st/Guineys   
9                          Talbot st/Murrays Pharmacy   
15              Westmoreland Street East/Fleet street   
18                  Westmoreland Street West/Carrolls   
34                                                NaN   
35                                                NaN   
36                                                NaN   
37                                                NaN   

                                   Footfall Locations  
4                              Aston Quay/Fitzgeralds  
21                       Bachelors walk/Bachelors way  
22                 Baggot st lower/Wilton tce inbound  
23                     Baggot st upper/Mespil rd/Bank  
3                                Capel st/Mary street  
12                      College Green/Bank Of Ireland  
19                          College Green/Church Lane  
20                         College st/Westmoreland st  
7                               D'olier st/Burgh Quay  
8                                  Dame Street/Londis  
16                Dawson Street/Molesworth Pedestrian  
11    Grafton Street / Nassau Street / Suffolk Street  
5                               Grafton Street/CompuB  
14                                 Grafton st/Monsoon  
24               Grand Canal st upp/Clanwilliam place  
25        Grand Canal st upp/Clanwilliam place/Google  
0                      Henry Street/Coles Lane/Dunnes  
17                        Liffey st/Halfpenny Bridge   
2                                   Mary st/Jervis st  
26           Newcomen Bridge/Charleville mall inbound  
27          Newcomen Bridge/Charleville mall outbound  
28         North Wall Quay/Samuel Beckett bridge East  
29         North Wall Quay/Samuel Beckett bridge West  
10                        O'Connell St/Parnell St/AIB  
13                     O'Connell St/Pennys Pedestrian  
1                       O'Connell st/Princes st North  
30                    Phibsborough Rd/Enniskerry Road  
31  Phibsborough Rd/Munster St (Removed due to Ove...  
32       Richmond st south/Portabello Harbour inbound  
33      Richmond st south/Portabello Harbour outbound  
6                                   Talbot st/Guineys  
9                   

Here we can see that dawson street, dawson street old, liffey street old, and o'connell street penneys temperary locations are unlisted. these locations were also among the counters listed for missing Data for both St.Patricks day, and most of the year. Hence, given minimal data, I will remove them from analysis going forward.

remove columns

In [31]:
footfall23_df = footfall23_df.drop(columns=['Dawson Street', 'Dawson Street old', 'Liffey Street old', "O'Connell Street Pennys - PYRO EVO Temporary Counter"])
patrick23_df = patrick23_df.drop(columns=['Dawson Street', 'Dawson Street old', 'Liffey Street old', "O'Connell Street Pennys - PYRO EVO Temporary Counter"])

1.3.1
Investigate the 1's...

In [32]:
null_rows = footfall23_df[footfall23_df['Mary st/Jervis st'].isnull()]
null_rows.head(1)

Aston Quay/Fitzgeralds  Bachelors walk/Bachelors way  \
Time                                                                        
2023-03-26 02:00:00                     NaN                           NaN   

                     Baggot st lower/Wilton tce inbound  \
Time                                                      
2023-03-26 02:00:00                                 NaN   

                     Baggot st upper/Mespil rd/Bank  Capel st/Mary street  \
Time                                                                        
2023-03-26 02:00:00                             NaN                   NaN   

                     College Green/Bank Of Ireland  College Green/Church Lane  \
Time                                                                            
2023-03-26 02:00:00                            NaN                        NaN   

                     College st/Westmoreland st  D'olier st/Burgh Quay  \
Time                                                                     
2023-03-26 02:00:00                         NaN                    NaN   

                     Dame Street/Londis  ...  O'Connell St/Pennys Pedestrian  \
Time                                     ...                                   
2023-03-26 02:00:00                 NaN  ...                             NaN   

                     O'Connell st/Princes st North  \
Time                                                 
2023-03-26 02:00:00                            NaN   

                     Phibsborough Rd/Enniskerry Road  \
Time                                                   
2023-03-26 02:00:00                              NaN   

                     Phibsborough Rd/Munster St (Removed due to Overcounting)  \
Time                                                                            
2023-03-26 02:00:00                                                NaN          

                     Richmond st south/Portabello Harbour inbound  \
Time                                                                
2023-03-26 02:00:00                                           NaN   

                     Richmond st south/Portabello Harbour outbound  \
Time                                                                 
2023-03-26 02:00:00                                            NaN   

                     Talbot st/Guineys  Talbot st/Murrays Pharmacy  \
Time                                                                 
2023-03-26 02:00:00                NaN                         NaN   

                     Westmoreland Street East/Fleet street  \
Time                                                         
2023-03-26 02:00:00                                    NaN   

                     Westmoreland Street West/Carrolls  
Time                                                    
2023-03-26 02:00:00                                NaN  

[1 rows x 34 columns]

All from the same time, perhaps a glitch in their recording system/system update... since these values increase and decrease, the values could be replaced by interpolation, backward fill, ML approach... I think interpolation by replacing the mean of the previous and next observation would be suitable  